### PR TITLE
abaddon: init at 0.1.10

### PR DESCRIPTION
**Pull request by GenericNerdyUsername**
###### Description of changes

abaddon is an alternative discord client.

https://github.com/uowuo/abaddon

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/abaddon/default.nix [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+, sqlite
+}:
+
+stdenv.mkDerivation rec {
+  pname = "abaddon";
+  version = "0.1.9";
+
+  src = fetchFromGitHub {
+    owner = "uowuo";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/applications/networking/instant-messengers/abaddon/default.nix:23]:
```suggestion
    repo = "abaddon";
```

File: pkgs/applications/networking/instant-messengers/abaddon/default.nix [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    rev = "v${version}";
+    hash = "sha256-Q6rBTG1WJRV0Oat/8fbs7mWEzcQdN+MztzmNEFDEqDw=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ cmake copyDesktopItems makeWrapper pkg-config ];
```
Comment (AndersonTorres) [inline pkgs/applications/networking/instant-messengers/abaddon/default.nix:29]:
```suggestion
  nativeBuildInputs = [ 
    cmake 
    copyDesktopItems 
    makeWrapper 
    pkg-config 
  ];

```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by GenericNerdyUsername** [comment]
Hows that?

File: pkgs/applications/networking/instant-messengers/abaddon/default.nix [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      mimeTypes = [ "x-scheme-handler/discord" ];
+    })
+  ];
+
+  meta = with lib; {
+    description = "A discord client reimplementation, written in C++";
+    homepage = "https://github.com/uowuo/abaddon";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ genericnerdyusername ];
+    platforms = [ "x86_64-linux" ];
```
Comment (AndersonTorres) [inline pkgs/applications/networking/instant-messengers/abaddon/default.nix:75]:
```suggestion
    platforms = lib.intersectLists lib.platforms.x86_64 lib.platforms.linux;
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/instant-messengers/abaddon/default.nix (hunk 3 above)
Comment (GenericNerdyUsername) [reply pkgs/applications/networking/instant-messengers/abaddon/default.nix]: should the `lib` prefix be dropped bc of the `with lib;`?

**Comment by GenericNerdyUsername** [comment]
the double semicolon is fixed now

File: pkgs/applications/networking/instant-messengers/abaddon/default.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/applications/networking/instant-messengers/abaddon/default.nix]: yes it can

**Comment by Aleksanaa** [comment]
Please update to `0.1.10`.

**Comment by GenericNerdyUsername** [comment]
done

**Comment by AndersonTorres** [comment]
1. Reword the title of PR
2. Ping me 2 days and I will merge this.

**Comment by AndersonTorres** [comment]
Reword the commit title too.

**Comment by GenericNerdyUsername** [comment]
@AndersonTorres

**Pull request merged**